src/golang/lib/lambda/utils.go

```diff
@@ -0,0 +1,223 @@
+package lambda
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/dropbox/godropbox/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type EcrAuth struct {
+	Username      string
+	Password      string
+	ProxyEndpoint string
+}
+
+func CreateLambdaFunction(functionType LambdaFunctionType, roleArn string) error {
+
+	// For each lambda function we create, we take the following steps:
+	// 1. Pull the image from the public ECR repository.
+	// 2. Create the private ECR repo if it doesn't exist
+	// 3. Get the ECR auth token and log in the docker client.
+	// 4. Push the image to the private ECR repo
+	// 5. Create the lambda function using the private ECR repo as the code.
+
+	lambdaImageUri, userRepoName, err := mapFunctionType(functionType)
+	if err != nil {
+		return errors.Wrap(err, "Unable to map function type to image.")
+	}
+	//TODO: REPLACE LATEST WITH IMAGE VERSION NUMBER
+	imageVersionNumber := "latest"
+
+	stdout := &bytes.Buffer{}
+	stderr := &bytes.Buffer{}
+
+	cmd := exec.Command("docker", "pull", fmt.Sprintf("%s:%s", lambdaImageUri, imageVersionNumber))
```

Comment:
How does this image version number compare to the ones that are found in the constants file?

Reply:
it should be the same value, I was unsure whether it was easier to set it as a constant or programmatically set it. Seemed easier if it was just one value we needed to change when doing a release.